src/swap/index.js

```diff
@@ -10,13 +11,111 @@ function getSwapContract(signer) {
   return new ethers.Contract(SWAP_CONTRACT_ADDRESS, abis[SWAP_CONTRACT_ADDRESS], signer)
 }
 
-function swap(order, signature, signer) {
+async function swap(orderParams, signer) {
+  const {
+    version,
+    signer: signerAddress,
+    r,
+    s,
+    v,
+    nonce,
+    makerWallet,
+    makerParam,
+    makerToken,
+    takerWallet,
+    takerParam,
+    takerToken,
+    expiry,
+  } = orderParams
+
+  const signature = {
+    version,
+    signer: signerAddress,
+    r,
+    s,
+    v,
+  }
+  const order = {
+    expiry,
+    nonce,
+    maker: { wallet: makerWallet.toLowerCase(), token: makerToken, param: makerParam },
+    taker: { wallet: takerWallet.toLowerCase(), token: takerToken, param: takerParam },
```

Comment:
Do you need a default taker wallet address here too?

Reply:
good catch! I do

Reply:
k this is handled in the order creation now, it will use the null address if none is specified